### ISSUE TITLE
[decorators] :after decoration breaks other overlapping decorations

### ISSUE THREAD

**Issue by siegebell** [Microsoft/vscode#8014]
- VSCode Version: 1.3.0-insider; commit a6aa320
- OS Version: Windows 8.1

When separately applying a decoration and an :after decoration to the same range of text, moving the cursor around will cause the former decoration to appear and disappear on all but the last character of the range. The effect is subtly dependent on the existence of lines of text elsewhere in the document; the bigger the document the more noticeable the bug gets.

I have not seen this bug with the :before decoration yet.

see: #7181

MWE:
```typescript
  const dec1 = vscode.window.createTextEditorDecorationType({
    textDecoration: "line-through",
  });
  const dec2 = vscode.window.createTextEditorDecorationType({
    after: { contentText: "$$$" }
  });
	function lineRange(line, start, end) { return new vscode.Range(line,start,line,end) }
  const editor = vscode.window.activeTextEditor;
  if (editor) {
    const lines = [
      "--- some-text --- (no decoration)",
      "--- some-text --- (line-through)",
      "--- some-text --- (line-through & :after)",
			"aaaaaa",
			"aaaaaa",
			"aaaaaa",
			"aaaaaa",
			"aaaaaa",
			"aaaaaa",
			"aaaaaa",
			"aaaaaa",
			"aaaaaa",
			"aaaaaa",
			"END OF EXAMPLE", "", "", ]
    editor.edit((edit) => {
      edit.insert(new vscode.Position(0,0), lines.join('\n'));
    }).then(() => {
      editor.setDecorations(dec1, [lineRange(1,4,13), lineRange(2,4,13)]);
      editor.setDecorations(dec2, [lineRange(2,4,13)]);
    });
  }
```

Observed bug (on & off, after clicking around a bit):
![before-after-plus-decoration-bug](https://cloud.githubusercontent.com/assets/16118166/16276786/e7a6edf4-387d-11e6-9faa-a310ee8c991e.PNG)

Expected:
![before-after-plus-decoration-expected](https://cloud.githubusercontent.com/assets/16118166/16276787/e7a7b9dc-387d-11e6-9794-dbd4847db617.PNG)

**Comment by dbaeumer** [comment]
@alexandrudima not sure if you or @aeschli

**Comment by aeschli** [comment]
duplicate of #10774

**Comment by aeschli** [comment]
@siegebell Thanks for the nice test case!
After the fix it looks like this:
![image](https://cloud.githubusercontent.com/assets/6461412/18065613/d869fd4e-6e34-11e6-94a0-00d204fa5326.png)

I realize that the after decoration inherits the text styles (strikethrough)...
